File: ArcherApp/app/src/main/java/me/mxcsyounes/archernotebook/database/converters/DateConverters.java
package me.mxcsyounes.archernotebook.database.converters;

import android.arch.persistence.room.TypeConverter;

import java.util.Date;

public class DateConverters {

    @TypeConverter
    public static Date toDate(Long time) {
        return time == null ? null : new Date(time);
    }

    @TypeConverter
    public static Long toLong(Date date) {
        return date == null ? null : date.getTime();
    }
}

File: ArcherApp/app/src/main/java/me/mxcsyounes/archernotebook/database/converters/DateConverters.kt
package me.mxcsyounes.archernotebook.database.converters

import android.arch.persistence.room.TypeConverter
import java.util.*

object DateConverters {

    @TypeConverter
    fun toDate(time: Long?): Date? {
        return if (time == null) null else Date(time)
    }

    @TypeConverter
    fun toLong(date: Date?): Long? {
        return date?.time
    }
}
